### DISCUSSION TITLE
Vote: Rename of devicePorts property of CreateSession object

### DISCUSSION THREAD

**Discussion by hdamker** [camaraproject/QualityOnDemand#142] [Polls]
Dear QoD team,

Please provide your vote on the question below.

For further background see the discussion in issue https://github.com/camaraproject/QualityOnDemand/issues/34 and PR https://github.com/camaraproject/QualityOnDemand/pull/139.

In the interests of transparency, and to attempt to maintain the one company one vote policy, I'd suggest that anyone who votes puts their vote in the comments.

Thanks,
Herbert

**Comment by ShutingQing** [comment]
When I read device local port, actually i need to think about what local means. Source port make more sense to me. :)

**Comment by patrice-conil** [comment]
Orange vote for deviceSourcePorts ... but to be honest  I don't know if it's clearer than deviceLocalPorts or devicePorts.

**Comment by eric-murray** [comment]
@patrice-conil, @ShutingQing 
Probably I should clarify why the property needs to be renamed.

Originally `devicePorts` was where the device public (NATed) port would be specified to identify the device (along with public IPv4 address specified in `device`). But the PR moves that into the DeviceIpv4Addr object.

So now the original `devicePorts` property is only required to identify ports for flow rules (both IPv4 and IPv6, though my opinion is that they are not actually required for IPv4). So not "public" ports, but the ports used directly by the device. Given the (subtle) change of use, a (subtle) change of name was considered expedient to highlight this.

Originally I proposed `deviceAllocatedPorts`, but @jlurien came up with the above suggestions, which I thought better. `devicePrivatePorts` was rejected as a possibility as this property also applies to IPv6 flows.

But if you have a better proposal, please put it forward and we can re-start the vote on this one.

**Reply by jlurien** [reply]
It is difficult to decide on a better name until we tackle how flows are to be fully specified. Will we need to define a flow direction?, in that case "source" ports may imply outgoing/uplink direction. Maybe "local" is more neutral then, or even `deviceFlowPorts`.
